packages/react-router/__tests__/path-matching-test.tsx

```diff
@@ -267,4 +275,123 @@ describe("path matching with splats", () => {
       pathnameBase: "/courses",
     });
   });
+
+  test("supports partial path matching with named parameters", () => {
+    let routes = [{ path: "/prefix:id" }];
+    expect(matchRoutes(routes, "/prefixabc")).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "params": Object {
+            "id": "abc",
+          },
+          "pathname": "/prefixabc",
+          "pathnameBase": "/prefixabc",
+          "route": Object {
+            "path": "/prefix:id",
+          },
+        },
+      ]
+    `);
+    expect(matchRoutes(routes, "/prefix/abc")).toMatchInlineSnapshot(`null`);
+  });
+
+  test("supports partial path matching with splat parameters", () => {
+    let routes = [{ path: "/prefix*" }];
+    expect(matchRoutes(routes, "/prefix/abc")).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "params": Object {
+            "*": "/abc",
+          },
+          "pathname": "/prefix/abc",
+          "pathnameBase": "/prefix",
+          "route": Object {
+            "path": "/prefix*",
+          },
+        },
+      ]
+    `);
+    expect(matchRoutes(routes, "/prefixabc")).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "params": Object {
+            "*": "abc",
+          },
+          "pathname": "/prefixabc",
+          "pathnameBase": "/prefix",
+          "route": Object {
+            "path": "/prefix*",
+          },
+        },
+      ]
+    `);
+  });
+});
+
+describe("route scoring", () => {
+  test.only("splat routes versus dynamic routes", () => {
+    let routes = [
+      { path: "nested/prefix-:param/static/prefix-*" }, // Score 43
+      { path: "nested/prefix-:param/static/*" }, // Score 33
+      { path: "nested/prefix-:param/static" }, // Score 34
+      { path: "nested/prefix-:param/*" }, // Score 22
+      { path: "nested/:param/static" }, // Score 28
+      { path: "nested/static" }, // Score 24
+      { path: "nested/prefix-:param" }, // Score 23
+      { path: "nested/prefix-*" }, // Score 22
+      { path: "nested/:param" }, // Score 17
+      { path: "static" }, // Score 13
+      { path: "prefix-:param" }, // Score 12
+      { path: "prefix-*" }, // Score 11
+      { path: ":param" }, // Score 6
+      { path: "*" }, // Score 1
```

Comment:
This is my attempt to pressure test all the potential combinations of partial/full splat/dynamic params and assert that we get the expected `matches`/`params`